qualtran/cirq_interop/t_complexity_protocol.py

```diff
@@ -31,6 +33,17 @@ class TComplexity:
     clifford: int = 0
     rotations: int = 0
 
+    def t_incl_rotations(self, eps: float = 1e-11) -> int:
+        """Return the total number of T gates after compiling rotations"""
+
+        # TODO Determine precise clifford count and/or ignore.
+        # This is an improvement over Ref. 2 from the docstring which provides
+        # a bound of 3 log(1/eps).
+        # See: https://github.com/quantumlib/Qualtran/issues/219
+        # See: https://github.com/quantumlib/Qualtran/issues/217
+        t_per_rot = int(np.ceil(1.149 * np.log2(1.0 / eps) + 9.2))
+        return self.t + t_per_rot * self.rotations
+
```

Comment:
This is fine as a first step, but we need to arrive at the right solution soon as highlighted here - https://github.com/quantumlib/Qualtran/issues/662#issuecomment-1949215534

I'm curious to see how quickly we can deprecate T-complexity in favour of Bloq counts. If it takes a while, I think it might be reasonable to update TComplexity to keep track of rotations as tuples in the meantime since it's such a fundamental part of cost analysis in the library right now.

Reply:
I mean you can already count rotations in that manner if you add them to the `keep` predicate